src/main/scala/Chisel/Module.scala

```diff
@@ -51,12 +51,22 @@ abstract class Module(_clock: Clock = null, _reset: Bool = null) extends HasId {
   def io: Bundle
   val clock = Clock(INPUT)
   val reset = Bool(INPUT)
+  /** globalReset is a simulation-only signal for ensuring assertions don't
+    * fire before the circuit comes out of reset
+    * It is the top-level Module's reset propagated throughout the circuit
+    */
+  val globalReset = _parent match {
+    case Some(p) => Bool(INPUT)
+    case None => reset
+  }
 
   private[Chisel] def addId(d: HasId) { _ids += d }
   private[Chisel] def ref = Builder.globalRefMap(this)
   private[Chisel] def lref = ref
 
-  private def ports = (clock, "clk") :: (reset, "reset") :: (io, "io") :: Nil
+  private def ports = (clock, "clk") :: (reset, "reset") :: (io, "io") ::
+    (if (_parent.nonEmpty && !this.isInstanceOf[BlackBox]) (globalReset, "globalReset") :: Nil
```

Comment:
globalReset is special because this is a private API and we don't want to force users to put globalReset as an input port in the Verilog for their blackboxes. We could require that because assertions in blackboxes could lead to the same problem, but I think they can just solve it in Verilog the Chisel2 way.